### PR TITLE
My Jetpack: Fix - Update getting raw post type count

### PR DESCRIPTION
**Pull request by oikeme**
Fixes # https://github.com/Automattic/jetpack/issues/38130

## Proposed changes:

* For sites with a high post count (over 1 million posts) a remote request will now be used instead of the local query found in get_raw_post_type_breakdown which has been identified as potentially causing backend issues.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* This test will require a site with over a million post to test that the remote call is triggered correctly. Alternatively this can be tested by updating the `POST_COUNT_QUERY_LIMIT` constant in the PR to a low number for testing purposes and to trigger the remote query.
* Visit My Jetpack and ensure that the remote call is made. You can verify this by confirming the option `raw_post_type_breakdown` is set on the remote site. 
* Ensure that My Jetpack loads as expected. 
* Verify the pricing page when clicking on `Get Search` loads as expected.

### CODE REVIEW

File: projects/packages/my-jetpack/src/products/class-search-stats.php [hunk 1]
```diff
@@ -36,6 +36,8 @@ class Search_Stats {
 	const CACHE_EXPIRY                  = 1 * MINUTE_IN_SECONDS;
 	const CACHE_GROUP                   = 'jetpack_search';
 	const POST_TYPE_BREAKDOWN_CACHE_KEY = 'post_type_break_down';
+	const TOTAL_POSTS_COUNT_CACHE_KEY   = 'total-post-count';
+	const POST_COUNT_QUERY_LIMIT        = 1e6;
```
Comment (gibrown) [inline projects/packages/my-jetpack/src/products/class-search-stats.php]: My only question is if we should lower this limit to something closer to 100k to make it less likely to cause even minor performance problems on slower hosting.

**Review by IanRamosC — DISMISSED** [review]
Other than @gibrown's question, the code looks good! 👍🏾

File: projects/packages/my-jetpack/src/products/class-search-stats.php (hunk 1 above)
Comment (oikeme) [reply projects/packages/my-jetpack/src/products/class-search-stats.php]: @gibrown this sounds reasonable to me and is actually the query limit we used to fix the other similar issue. I did some testing to try to determine where things really start to break down in extreme cases but I think you're right and erring on the side of caution always makes sense so I've updated this to 100k.

**Review by IanRamosC — APPROVED** [review]

**Comment by kangzj** [comment]
Thanks for fixing this @oikeme 👍

**Pull request merged**